tests/SpecTests/SearchIndexSpecTest.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests;
+
+use Closure;
+use MongoDB\Collection;
+use MongoDB\Model\CachingIterator;
+use MongoDB\Tests\FunctionalTestCase;
+
+use function bin2hex;
+use function count;
+use function hrtime;
+use function iterator_to_array;
+use function json_encode;
+use function random_bytes;
+use function sleep;
+use function sprintf;
+
+use const JSON_THROW_ON_ERROR;
+
+/**
+ * Functional tests for the Atlas Search index management.
+ *
+ * @see https://github.com/mongodb/specifications/blob/master/source/index-management/tests/README.rst#search-index-management-helpers
+ * @group atlas
+ */
+class SearchIndexSpecTest extends FunctionalTestCase
+{
+    private const WAIT_TIMEOUT_SEC = 300;
+
+    public function setUp(): void
+    {
+        if (! self::isAtlas()) {
+            self::markTestSkipped('Search Indexes are only supported on MongoDB Atlas 7.0+');
+        }
+
+        parent::setUp();
```

Comment:
Is the parent method called in the middle because `isAtlas()` does not depend on `$this->manager`, while `skipIfServerVersion()` does? Just want to confirm, as tests typically just lead with the parent method.